### PR TITLE
Modify documentation in core/include/context

### PR DESCRIPTION
**Pull request by HalynaDumych**
Relates-to: OLPEDGE-1442

Signed-off-by: Halyna Dumych <ext-halyna.dumych@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/include/olp/core/context/Context.h [hunk 1]
```diff
@@ -33,57 +33,108 @@
...
  public:
+  /**
+   * @brief Called when the `Context` object is initialized.
+   *
+   * Controlled by the `Scope` class.
+   *
+   * @see `Scope` for more information.
+   */
   using InitializedCallback = std::function<void()>;
+  /**
```
Comment (r0busta) [inline olp-cpp-sdk-core/include/olp/core/context/Context.h]:
```suggestion

  /**
```

File: olp-cpp-sdk-core/include/olp/core/context/Context.h [hunk 2]
```diff
@@ -33,57 +33,108 @@
...
+   * Instead of calling `Context::init()` and `Context::deinit()` manually,
+   * instantiate the `Context::Scope()` object.
    */
   class CORE_API Scope {
    public:
-    /// see Context::init()
+    /**
+     * @brief Creates the `Scope` instance.
+     *
+     * Used to initialize the `Context` class.
```
Comment (r0busta) [inline olp-cpp-sdk-core/include/olp/core/context/Context.h]: `<rant>` I see such sentences starting with `Used to ...` all across the SDK and while they might make sense in the library context. They sound weird, and I'm not even sure that such a sentence is grammatically correct.`</rant>`
What do you think should fix it some other day?

Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/context/Context.h]: In the descriptions, it is a common practice not to repeat the class/method/function name and not to use phrases like 'this method", "this function", and so on. This sentence is a shorter version of  "This constructor is used to initialize the `Context` class". 

For more information, please see the [API Reference Code Comments](https://developers.google.com/style/api-reference-comments) section in the 
Developer Documentation Style Guide.

Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/context/Context.h]: However, we can put it in the following way: "The `Scope` instance is used to initialize the `Context` class."

File: olp-cpp-sdk-core/include/olp/core/context/Context.h (hunk 1 above)
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/context/Context.h]: Added

File: olp-cpp-sdk-core/include/olp/core/context/Context.h (hunk 2 above)
Comment (r0busta) [reply olp-cpp-sdk-core/include/olp/core/context/Context.h]: I was just asking a question, but thanks for changing that quick and pro-actively anyway :)

**Pull request merged**